libraries/ESP8266WiFi/src/WiFiClient.h

```diff
@@ -84,6 +84,13 @@ class WiFiClient : public Client, public SList<WiFiClient> {
   static void stopAll();
   static void stopAllExcept(WiFiClient * c);
 
+  void keepAlive (uint16_t idle_sec, uint16_t intv_sec, uint8_t count);
+  uint16_t getIdleKeepAlive ();
+  void disableKeepAlive () { keepAlive(0, 0, 0); }
+
+  static void setKeepAlive (void* pcb, uint16_t idle_sec, uint16_t intv_sec, uint8_t count);
+  static void getKeepAlive (void* pcb, uint16_t* idle_sec, uint16_t* intv_sec, uint8_t* count);
```

Comment:
How about moving these two into ClienContext? Exposing details such as pcb does not look right...
If you move it into ClienContext, then the first argument can also be changed from void* to tcp_pcb*.

Reply:
WiFiClient.cpp has <lwip/*.h>
ClientContext has private _pcb
You make me realize that since ClientContext knows tcp_pcb, it has <lwip/*.h> too.
will fix :)